bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/state/PowermaxState.java

```diff
@@ -614,62 +289,34 @@ private static String getShortArmMode(String armMode) {
-                    && getSensorLastTripped(i).equals(otherState.getSensorLastTripped(i))) {
-                setSensorLastTripped(i, null);
-            }
-            if ((isSensorLowBattery(i) != null) && isSensorLowBattery(i).equals(otherState.isSensorLowBattery(i))) {
-                setSensorLowBattery(i, null);
-            }
-            if ((isSensorBypassed(i) != null) && isSensorBypassed(i).equals(otherState.isSensorBypassed(i))) {
-                setSensorBypassed(i, null);
-            }
-            if ((isSensorArmed(i) != null) && isSensorArmed(i).equals(otherState.isSensorArmed(i))) {
-                setSensorArmed(i, null);
+        for (int zone = 1; zone <= zones.length; zone++) {
+            PowermaxZoneState thisZone = getZone(zone);
+            PowermaxZoneState otherZone = otherState.getZone(zone);
+
+            for (int i = 0; i < thisZone.getValues().size(); i++) {
+                Value<?> thisValue = thisZone.getValues().get(i);
+                Value<?> otherValue = otherZone.getValues().get(i);
+
+                if ((thisValue.getValue() != null) && thisValue.getValue().equals(otherValue.getValue())) {
+                    thisValue.setValue(null);
+                }
             }
         }
+
         for (int i = 0; i < pgmX10DevicesStatus.length; i++) {
             if ((getPGMX10DeviceStatus(i) != null)
                     && getPGMX10DeviceStatus(i).equals(otherState.getPGMX10DeviceStatus(i))) {
                 setPGMX10DeviceStatus(i, null);
             }
         }
-        if ((ready != null) && ready.equals(otherState.isReady())) {
-            ready = null;
-        }
-        if ((bypass != null) && bypass.equals(otherState.isBypass())) {
-            bypass = null;
-        }
-        if ((alarmActive != null) && alarmActive.equals(otherState.isAlarmActive())) {
-            alarmActive = null;
-        }
-        if ((trouble != null) && trouble.equals(otherState.isTrouble())) {
-            trouble = null;
-        }
-        if ((alertInMemory != null) && alertInMemory.equals(otherState.isAlertInMemory())) {
-            alertInMemory = null;
-        }
-        if ((statusStr != null) && statusStr.equals(otherState.getStatusStr())) {
-            statusStr = null;
-        }
-        if ((armMode != null) && armMode.equals(otherState.getArmMode())) {
-            armMode = null;
-        }
-        if ((lastKeepAlive != null) && lastKeepAlive.equals(otherState.getLastKeepAlive())) {
-            lastKeepAlive = null;
-        }
-        if ((panelStatus != null) && panelStatus.equals(otherState.getPanelStatus())) {
-            panelStatus = null;
-        }
-        if ((alarmType != null) && alarmType.equals(otherState.getAlarmType())) {
-            alarmType = null;
-        }
-        if ((troubleType != null) && troubleType.equals(otherState.getTroubleType())) {
-            troubleType = null;
+
+        for (int i = 0; i < getValues().size(); i++) {
```

Comment:
For powerlinkMode and downloadMode, it's unnecessary maybe but not harmful, as they'll get set initially and never change. I guess this is why keepOnlyDifferencesWith() and merge() just ignore them, because they're not expected to change.

For downloadSetupRequired, I'll need to fix this. It's used in PowermaxBridgeHandler.onNewStateEvent(), directly from the new state and it's not merged into the persistent state... which means it never gets set to false after the download is done. I think the right behavior is to just set it to false and otherwise be consistent with everything else.

The dynamic ones are no issue since their setters are ignored.